tests/robustness/report/wal.go

```diff
@@ -356,3 +345,56 @@ func toEtcdOperation(op *pb.RequestOp) (operation model.EtcdOperation) {
 	}
 	return operation
 }
+
+func ReadAllWALEntries(lg *zap.Logger, dirpath string) (state raftpb.HardState, ents []raftpb.Entry, err error) {
+	names, err := fileutil.ReadDir(dirpath)
+	if err != nil {
+		return state, nil, err
+	}
+	files := make([]fileutil.FileReader, 0, len(names))
+	for _, name := range names {
+		if !strings.HasSuffix(name, ".wal") {
+			continue
+		}
+		p := filepath.Join(dirpath, name)
+		var f *os.File
+		f, err = os.OpenFile(p, os.O_RDONLY, fileutil.PrivateFileMode)
+		if err != nil {
+			return state, nil, fmt.Errorf("os.OpenFile failed (%q): %w", p, err)
+		}
+		defer f.Close()
+		files = append(files, fileutil.NewFileReader(f))
+	}
+	rec := &walpb.Record{}
+	decoder := wal.NewDecoder(files...)
+	for err = decoder.Decode(rec); err == nil; err = decoder.Decode(rec) {
+		switch rec.Type {
+		case wal.EntryType:
+			e := wal.MustUnmarshalEntry(rec.Data)
+			i := len(ents)
+			for ; i > 0 && ents[i-1].Index >= e.Index; i-- {
```

Comment:
Nit: would this work the same way?
```suggestion
   //  https://pkg.go.dev/sort#Search  - just for your reference, not intended for the code
   i = sort.Search(len(ents), func(i int) bool {
     ents[i].Index == e.Index
   })
```